src/preload-hover.js

```diff
@@ -19,17 +19,23 @@ export default class PreloadHover {
 
     domScopes.forEach(domScope => {
       const links = [...domScope.getElementsByTagName('a')];
-
+      let timer;
       let uniqueLinks = uniqBy(links, 'href');
 
       uniqueLinks.forEach(link => {
         link.addEventListener('mouseover', () => {
-          const preload = document.createElement('link');
-
-          preload.setAttribute('rel', 'preload');
-          preload.setAttribute('href', link.href);
+          timer = setTimeout(() =>{
```

Comment:
This cannot reuses the same `timer` before the existing one is completed. Otherwise, a `mouseout` could be clearing the timer for a *different* `a` tag.